x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportGetDataFrameAnalyticsStatsAction.java

```diff
@@ -157,75 +160,99 @@ void gatherStatsForStoppedTasks(List<String> expandedIds, GetDataFrameAnalyticsS
             return;
         }
 
-        searchStoredProgresses(stoppedTasksIds, ActionListener.wrap(
-            storedProgresses -> {
-                List<Stats> stoppedStats = new ArrayList<>(stoppedTasksIds.size());
-                for (int i = 0; i < stoppedTasksIds.size(); i++) {
-                    String configId = stoppedTasksIds.get(i);
-                    StoredProgress storedProgress = storedProgresses.get(i);
-                    stoppedStats.add(buildStats(configId, storedProgress.get()));
-                }
-                List<Stats> allTasksStats = new ArrayList<>(runningTasksResponse.getResponse().results());
-                allTasksStats.addAll(stoppedStats);
-                Collections.sort(allTasksStats, Comparator.comparing(Stats::getId));
-                listener.onResponse(new GetDataFrameAnalyticsStatsAction.Response(new QueryPage<>(
-                    allTasksStats, allTasksStats.size(), GetDataFrameAnalyticsAction.Response.RESULTS_FIELD)));
-            },
-            listener::onFailure
-        ));
+        AtomicInteger counter = new AtomicInteger(stoppedTasksIds.size());
+        AtomicArray<Stats> jobStats = new AtomicArray<>(stoppedTasksIds.size());
+        for (int i = 0; i < stoppedTasksIds.size(); i++) {
```

Comment:
I wonder if we will hit scaling issues if there are 100s of stopped tasks. 

Seems like we could be making 100s of unbatched, search requests.

Reply:
This is why I changed the code to make a multi-search per job. If we batch everything up, then we have jobs * stats_fields searches in a single multi-search and that to have its own problems. We do it this way for anomaly detection jobs too. Not sure there's a better way.